File: 프로그래머스/Level 2/이진 변환 반복하기.py
# 이진 변환 반복하기
# https://school.programmers.co.kr/learn/courses/30/lessons/70129
# 0제거
def removeZero(st):
    cnt = 0
    for s in st:
        if s == "0":
            cnt += 1
    return cnt

# 이진 변환


def tranBin(num):
    ret = ""

    while num > 0:
        ret += str(num % 2)
        num //= 2

    return ret[::-1]


def solution(s):
    answer = [0, 0]
    st = s

    while st != "1":
        answer[0] += 1  # 회차 시작
        # 0 제거
        answer[1] += removeZero(st)
        ss = "".join(st.split("0"))
        # 2진수 변환
        binStr = tranBin(len(ss))
        st = binStr
    return answer

File: 프로그래머스/Level 2/이진 변환 반복하기.swift
// https://school.programmers.co.kr/learn/courses/30/lessons/70129?language=swift
import Foundation

func solution(_ s:String) -> [Int] {
    var answer:[Int] = [0,0]
    var s = s
    
    func removeZero(_ x: [String]) -> [String] {
        var ret:[String] = []
        
        for i in 0..<x.count {
            if x[i] != "0" {
                ret.append(x[i])
            } else {
                answer[1] += 1
            }
        }
        
        return ret
    }
    
    func getBinaryNumber(_ c: Int) -> [String] {
        var ret:[String] = []
        var number:Int = c
        
        while number > 0 {
            ret.append(String(number%2))
            number /= 2
        }
        
        return ret.reversed()
    }
    
    
    while true {
        if s == "1" {
            break
        }
        answer[0] += 1
        
        let removedZero = removeZero(s.map {String($0)})
        s = getBinaryNumber(removedZero.count).joined()
    }
    
    return answer
}